aws/logs_monitoring/lambda_function.py

```diff
@@ -370,15 +386,22 @@ def send(self, logs):
         """
         Sends a batch of log, only retry on server and network errors.
         """
+        try:
+            data=self._scrubber.scrub("[{}]".format(",".join(logs)))
+        except ScrubbingException:
+            raise Exception("could not scrub the payload")
+        if DD_USE_COMPRESSION:
+            try:
+                data = compress_logs(str(data), DD_COMPRESSION_LEVEL)
```

Comment:
Why do we need to convert the data to a string here ?

Reply:
I think we'll recompress the same batch over and over on retries, we could probably do the compression outside of this method to avoid extra computations on network errors.

Reply:
I went ahead and removed the try block.

For the casting to string, I was getting some type errors for a while from the `gzip.compress` function when trying to convert `data` to bytes and casting it to a string first seemed to fix it. I just tried again though without it and everything works ok. Must have been some other error I fixed inadvertently. *shrug*